### DISCUSSION TITLE
Alert Message States and Silences retention periods should be configurable.

### DISCUSSION THREAD

**Discussion by joeblubaugh** [grafana/grafana#46676] [Alerting]
As part of the discussion of https://github.com/grafana/grafana/issues/41156, @armandgrillet mentioned a feature request:
We should make the retention period for expired silences and message states.

This FR proposes a single retention period for both expiring message states and silences configurable.

I'd like to ask for community feedback:
1. Do you want retention for these types in the Grafana backend to be configurable? Please react 🎉  
2. If yes on 1, would you be happy with a single setting for both message states and silences? Please react 🚀 

If yes on 1 and no on 2, I would love it if you commented with a use-case that supports different retention settings per type.

**Comment by grimz-ly** [comment]
I was actually unaware of the retention set on message states. Guessing it is currently 5 as well?

Anyhow, 5 days is entirely too short for our environment. If static, 30 days would be OK and 45 days ideal for recurring monthly tasks where we want a tailored silence to be recreated easily. Configurable would be nice for something needed less frequent, quarterly.

Thank you for considering this.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
